pkg/config/configuration.go

```diff
@@ -139,6 +142,22 @@ type APIAccessRule struct {
 	Protocol string `json:"protocol"`
 }
 
+// APIAccessRules is a list of API access rules (allowlist or denylist).
+type APIAccessRules []APIAccessRule
+
+// ForProtocol returns a list of APIAccessRule objects filtered by protocol
+func (r APIAccessRules) ForProtocol(protocol string) []APIAccessRule {
+	res := make([]APIAccessRule, len(r))
```

Comment:
how about `func (r APIAccessRules) ForProtocol(protocol string) (ret []APIAccessRule) {` to reduce `n`?